app/src/DatasetExplorer.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import DatasetFilter from "./DatasetFilter";
+import DataFetcher from "./DataFetcher";
+import DataTable from "./DataTable";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import DatasetListing from "./DatasetListing";
+import styles from "./DatasetExplorer.module.scss";
+import ChevronRightIcon from "@material-ui/icons/ChevronRight";
+
+const sources = [
+  {
+    id: "state_names",
+    displayName: "State Names",
+    description: "List of states and their FIPS codes.",
+  },
+  {
+    id: "county_names",
+    displayName: "County Names",
+    description: "List of counties and their FIPS codes.",
+  },
+  {
+    id: "pop_by_race",
+    displayName: "County Population by Race",
+    description: "The population of each county broken down by race.",
+  },
+];
+
+type LoadStatus = "unloaded" | "loading" | "loaded";
+
+function DatasetExplorer() {
+  const [loadStatus, setLoadStatus] = useState<LoadStatus>("unloaded");
+  const [previewedSourceId, setPreviewedSourceId] = useState("");
+  const [data, setData] = useState([]);
+  const [columns, setColumns] = useState([]);
+  const [activeFilter, setActiveFilter] = useState<Array<string>>([]);
+
+  const loadPreview = async (sourceId: string) => {
+    setLoadStatus("loading");
+    setPreviewedSourceId(sourceId);
+    const fetcher = new DataFetcher();
+    // TODO error handling
+    const source = await fetcher.loadDataset(sourceId);
+    setData(source.data);
+    setColumns(source.columns);
+    setLoadStatus("loaded");
+  };
+
+  function filterSources(filtered: Array<string>) {
+    setActiveFilter(filtered);
+  }
+
+  let table;
```

Comment:
What do you think about inlining this in the render() ? I prefer doing this because I think it makes it easier to parse rather than having to remember what's in various variables if you come to have a lot

```
<div className={styles.Table}>
          {(() => {
            switch (loadStatus) {
              case 'loaded':
                return <DataTable columns={columns} data={data} />;
              case 'loading':
                return <p>Loading...</p>;
              default:
                return <p>Select a data source to view.</p>;
            }
          })()}
        </div>
```

Reply:
I sort of disagree here, I think having it extracted as a separate function makes it more readable. I think a lot of React style guides prefer the latter as well, but it's also subjective. Might be worth coming to consensus on in terms of a style guide.

Reply:
I guess I'm also a fan of naming functions in a way that help explain their intent, e.g. something like `function renderTableOrMsg() {}` or `maybeRenderTable() {}`.

```js
function renderTableOrMsg() {
  switch (loadStatus) {
    case 'loaded':
      return <DataTable columns={columns} data={data} />;
    case 'loading':
      return <p>Loading...</p>;
    default:
      return <p>Select a data source to view.</p>;
  }
}
```

Reply:
This is probably my least favorite part of React (especially with functional components and hooks) is it encourages mega functions rather than breaking things up into readable helper functions :/

I very definitely lean towards Chris's solution of extracting to a helper function in situations like this, assuming that the helper function can be named in a reasonable way, if that's okay with you Krista.